### PR TITLE
Configure pre-commit

### LINKED ISSUE #16: ⚙ Configure pre-commit tool
We can now start configuring linting-related things.

Please take as a base these:
- https://github.com/ansible/pylibssh/blob/devel/.pre-commit-config.yaml
- https://github.com/ansible/ansible-lint/blob/master/.pre-commit-config.yaml#L37-L43
- https://github.com/ansible/ansible-lint/blob/master/.yamllint (but turn `consistent` to `false`)
- https://github.com/ansible/pylibssh/blob/devel/.darglint
- https://github.com/ansible/pylibssh/blob/devel/.isort.cfg
- https://github.com/ansible/pylibssh/blob/devel/.flake8
- https://github.com/ansible/ansible-lint/blob/master/mypy.ini / https://github.com/sanitizers/octomachinery/blob/master/mypy.ini / https://github.com/ansible/ansible-lint/blob/master/.pre-commit-config.yaml#L60-L67
- https://github.com/ansible/ansible-lint/blob/master/.pre-commit-config.yaml#L68-L71 / https://github.com/sanitizers/octomachinery/blob/master/.pre-commit-config.yaml#L51-L58 / https://github.com/sanitizers/octomachinery/blob/master/.pylintrc (generate the base config via `pylint --generate-rcfile > .pylintrc` and only then borrow things from examples, only if necessary, in separate commits, for comparison)

Since it's a lot of changes, feel free to submit separate PRs with small changes.

### PR DESCRIPTION
**Pull request by slsh1o**
Looks like it would be simpler just add different hooks via several commits (like with yamllint) in this PR than create several others PRs.

Ref #16

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+- repo: https://gitlab.com/pycqa/flake8.git
+  rev: 3.8.2
+  hooks:
+  - id: flake8
+    language_version: python3
+    additional_dependencies:
+    - flake8-2020>=1.6.0
+    - flake8-docstrings
+    - flake8-pytest-style>=1.0.0
+    - isort<5  # https://github.com/gforcada/flake8-isort/issues/88
```
Comment (webknjaz) [inline .pre-commit-config.yaml:63]: This restriction is no longer necessary
```suggestion
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
+---
+repos:
+- repo: git://github.com/timothycrosley/isort
+  rev: 5.1.1
+  hooks:
+  - id: isort
+    args:
+    - --honor-noqa
+    files: >-
+      ^.*\.p(xd|y|yx)$
```
Comment (webknjaz) [inline .pre-commit-config.yaml:10]: this is unnecessary because the project is pure-python
```suggestion
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  hooks:
+  - id: pydocstyle
+
+- repo: https://github.com/adrienverge/yamllint.git
+  rev: v1.24.2
+  hooks:
+  - id: yamllint
+    files: \.(yaml|yml)$
+    types: [file, yaml]
+    entry: yamllint --strict
```
Comment (webknjaz) [inline .pre-commit-config.yaml:86]: I think it's better to specify args:
```suggestion
    args:
    - --strict
```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
+---
+repos:
+- repo: git://github.com/timothycrosley/isort
```
Comment (webknjaz) [inline .pre-commit-config.yaml:3]: I think it's better to submit this along with `.isort.cfg`

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  rev: 3.8.2
+  hooks:
+  - id: flake8
+    language_version: python3
+    additional_dependencies:
+    - flake8-2020>=1.6.0
+    - flake8-docstrings
+    - flake8-pytest-style>=1.0.0
+    - isort<5  # https://github.com/gforcada/flake8-isort/issues/88
+    - wemake-python-styleguide
```
Comment (webknjaz) [inline .pre-commit-config.yaml:64]: Does this really not generate any complaints?

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    files: >-
+      ^.*\.p(xd|y|yx)$
```
Comment (webknjaz) [inline .pre-commit-config.yaml:66]: Cython support isn't really necessary here
```suggestion
```

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  - id: detect-private-key
+
+  # Heavy checks:
+  - id: check-ast
+  - id: debug-statements
+
+- repo: https://gitlab.com/pycqa/flake8.git
+  rev: 3.8.2
+  hooks:
+  - id: flake8
```
Comment (webknjaz) [inline .pre-commit-config.yaml:57]: It's better to submit this with `.flake8` or even submit that config first.

**Comment by slsh1o** [comment]
I've deleted isort and flake8, should I add them here in next commits or would be better to add them in separate PR?

File: .pylintrc [hunk 8]
```diff
@@ -0,0 +1,595 @@
...
+evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / statement) * 10)
+
+# Template used to display messages. This is a python new-style format string
+# used to format the message information. See doc for all details.
+#msg-template=
+
+# Set the output format. Available formats are text, parseable, colorized, json
+# and msvs (visual studio). You can also give a reporter class, e.g.
+# mypackage.mymodule.MyReporterClass.
+output-format=text
```
Comment (webknjaz) [inline .pylintrc:166]: this should probably be colorized
```suggestion
output-format=colorized
```

File: src/sphinxcontrib/towncrier/_version.py [hunk 9]
```diff
@@ -1,6 +1,7 @@
 """Version definition."""
 
 try:
+    # pylint: disable=unused-import
     from ._scm_version import version as __version__
```
Comment (webknjaz) [inline src/sphinxcontrib/towncrier/_version.py:5]: It's preferable to put the ignores on the same line they are applied against:
```suggestion
    from ._scm_version import version as __version__  # pylint: disable=unused-import
```

Comment (webknjaz) [reply src/sphinxcontrib/towncrier/_version.py:5]: Oh, it'd be too long. Nevermind, then...

File: .pylintrc [hunk 10]
```diff
@@ -0,0 +1,595 @@
...
+
+# Number of spaces of indent required inside a hanging or continued line.
+indent-after-paren=4
+
+# String used as indentation unit. This is usually "    " (4 spaces) or "\t" (1
+# tab).
+indent-string='    '
+
+# Maximum number of characters on a single line.
+max-line-length=100
```
Comment (webknjaz) [inline .pylintrc:234]: I believe this should be
```suggestion
max-line-length=79
```

**Pull request merged**